src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -945,8 +945,17 @@ public void Connect(string host, int port)
                 throw new NotSupportedException(SR.net_invalidversion);
             }
 
-            IPAddress[] addresses = Dns.GetHostAddressesAsync(host).GetAwaiter().GetResult();
-            Connect(addresses, port);
+            IPAddress parsedAddress;
```

Comment:
Windows doesn't have this problem: I believe this needs to be done within the PAL layer.